taxi/tests/tests_view.py

```diff
@@ -0,0 +1,108 @@
+    def setUp(self):
+        self.user = get_user_model().objects.create_user(
+            username="admin",
+            password="admin12345"
+        )
+
+        self.client.force_login(self.user)
+
+    def test_retrieve_manufacturers(self):
+        Manufacturer.objects.create(name="Toyota", country="Japan")
+        Manufacturer.objects.create(name="Nissan", country="Japan")
+        response = self.client.get(MANUFACTURER_URL)
+
+        manufacturers = Manufacturer.objects.all()
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(
+            list(response.context["manufacturer_list"]),
+            list(manufacturers)
+        )
+        self.assertTemplateUsed(response, "taxi/manufacturer_list.html")
+
+    def test_manufacturer_search_field(self):
+        response = self.client.get(MANUFACTURER_URL + "?name=Toyota")
+        expected_queryset = Manufacturer.objects.filter(
+            name__icontains="Toyota"
+        )
+
+        self.assertQuerysetEqual(
+            response.context["manufacturer_list"],
+            expected_queryset
+        )
+
+
+class PrivateDriverTests(TestCase):
+    def setUp(self) -> None:
+        self.user = get_user_model().objects.create_user(
+            "test",
+            "password123"
+        )
+        self.client.force_login(self.user)
+
+    def test_create_driver(self):
+        form_data = {
+            "username": "test_username",
+            "license_number": "AAA12345",
+            "first_name": "test_first_name",
+            "last_name": "test_last_name",
+            "password1": "test_password",
+            "password2": "test_password"
+        }
+        self.client.post(reverse("taxi:driver-create"), data=form_data)
+        new_user = get_user_model().objects.get(username=form_data["username"])
+
+        self.assertEqual(new_user.first_name, form_data["first_name"])
+        self.assertEqual(new_user.last_name, form_data["last_name"])
+        self.assertEqual(new_user.license_number, form_data["license_number"])
+
+
+class ToggleAssignToCarViewTest(TestCase):
+    def setUp(self):
+        self.driver = Driver.objects.create_user(
+            username="user",
+            password="TES12345"
+        )
+        self.manufacturer = Manufacturer.objects.create(
+            name="Test Manufacturer"
+        )
+        self.car = Car.objects.create(
+            model="Test car model",
+            manufacturer=self.manufacturer
+        )
+
+    def test_toggle_assign_to_car(self):
+        self.assertFalse(self.car in self.driver.cars.all())
+
+        self.client.force_login(self.driver)
+        response = self.client.post(reverse(
+            "taxi:toggle-car-assign",
+            args=[self.car.id])
+        )
+        self.assertEqual(response.status_code, 302)
+        self.assertTrue(self.car in self.driver.cars.all())
+
+        response = self.client.post(reverse(
+            "taxi:toggle-car-assign",
```

Comment:
Why do you have only `MANUFACTURER_URL` constant ?

Reply:
Because I used reverse("taxi:manufacturer-list")  more than once or twice for method GET
At the same time, in the case reverse("taxi:toggle-car-assign") I had to add an extra argument to the link which didn`t allow me to move this link into a separate constant (I tried several options, all unsuccessful).
So I have only one constant here - MANUFACTURER_URL.